ballerina/yaml_api.bal

```diff
@@ -38,16 +94,55 @@ public enum YAMLSchema {
 }
 
 # Represent the options that can be used for filtering in the projection.
-#
-# + schema - field description  
-# + allowAnchorRedefinition - field description  
-# + allowMapEntryRedefinition - field description
-public type Options record {
+public type Options record {|
+    # + schema - field description
     YAMLSchema schema = CORE_SCHEMA;
+    # + allowAnchorRedefinition - field description
     boolean allowAnchorRedefinition = true;
+    # + allowMapEntryRedefinition - field description
     boolean allowMapEntryRedefinition = false;
-};
+    # + allowDataProjection - Enable or disable projection
+    record {
+        # If `true`, nil values will be considered as optional fields in the projection.
+        boolean nilAsOptionalField = false;
+        # If `true`, absent fields will be considered as nilable types in the projection.
+        boolean absentAsNilableType = false;
+        # If `true`, top level tuple ordering considered strictly.
+        boolean strictTupleOrder = false;
```

Comment:
As default we reorder the YAML stream right? Is this the option to disable reorder?

Hence shall we rename it to convey that meaning like `enableStreamReorder`

Reply:
Yes, this name is better.